src/Raven.Studio/typescript/viewmodels/database/query/query.ts

```diff
@@ -635,8 +634,28 @@ class query extends viewModelBase {
         if (!indexName)
             return null;
 
-        const indexes = this.indexes() || [];
-        return indexes.find(i => i.Name === indexName);
+        let currentIndex = this.indexes() ? this.indexes().find(i => i.Name === indexName) : null;
+        
+        if (currentIndex) {
+            return currentIndex;
+        }
+        
+        // fetch indexes since this view may not be up-to-date if index was defined outside of studio
+        const deferred = $.Deferred();
```

Comment:
create $.Deferred<string> and resolve with proper value - instead of creating variable for that.